src/main/java/com/matchbook/sdk/clients/rest/UserResponseImpl.java

```diff
@@ -0,0 +1,74 @@
+package com.matchbook.sdk.clients.rest;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.databind.ObjectReader;
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.matchbook.sdk.StreamObserver;
+import com.matchbook.sdk.clients.rest.dtos.user.LoginRequest;
+import com.matchbook.sdk.clients.rest.dtos.user.LoginResponse;
+import com.matchbook.sdk.clients.rest.dtos.user.LogoutResponse;
+import com.matchbook.sdk.configs.ClientConnectionManager;
+import com.matchbook.sdk.exceptions.MatchbookSDKHTTPException;
+import com.squareup.okhttp.Callback;
+import com.squareup.okhttp.MediaType;
+import com.squareup.okhttp.Request;
+import com.squareup.okhttp.RequestBody;
+import com.squareup.okhttp.Response;
+import com.squareup.okhttp.ResponseBody;
+
+public class UserResponseImpl implements UserResource {
+
+    private final MediaType mediaType;
+    private final ClientConnectionManager clientConnectionManager;
+    private final String loginURL;
+    private final ObjectReader loginResponseReader;
+    private final ObjectWriter loginResponseWriter;
+
+    public UserResponseImpl(ClientConnectionManager clientConnectionManager) {
+        this.clientConnectionManager = clientConnectionManager;
+        this.loginResponseReader = clientConnectionManager.getMapper().readerFor(LoginResponse.class);
+        this.loginResponseWriter = clientConnectionManager.getMapper().writerFor(LoginRequest.class);
+        this.loginURL = clientConnectionManager.getClientConfig().getLoginUrl();
+        this.mediaType = MediaType.parse("application/json");
+    }
+
+    @Override
+    public void login(LoginRequest loginRequest, StreamObserver<LoginResponse> observer) {
+        try {
+            RequestBody requestBody
+                    = RequestBody.create(mediaType, loginResponseWriter.writeValueAsString(loginRequest));
+
+            Request request = new Request.Builder()
+                    .url(loginURL)
+                    .addHeader("Accept", "application/json")
+                    .post(requestBody)
+                    .build();
+
+            clientConnectionManager.getHttpClient().newCall(request).enqueue(new Callback() {
```

Comment:
That's fine for now, but as a note for future reference, we can hide the OKHttp library usages in an internal wrapper in order to make the Matchbook SDK more flexible, extend the `Callback` interface in a custom interface, define the base behaviour in an abstract implementation (e.g. the failure case) and leverage the template method pattern to handle the successful responses.

Reply:
Yep, i completely agree with you! I did that atm as a placeholder, so we agree on a main design flow